Create OpenD executable (#15)
opendlang/opend: opend/README.md

# OpenD

This is the main page for the OpenD language build system.  

> **WARNING** This is still work in progress, and in very early stages of development. Expect stuff to change rapidly.

Reviewer: ```suggestion
**WARNING**: This is still a work in progress, and in very early stages of development. Expect stuff to change rapidly.
```

## Meta

> The main idea is, calling `opend` should "Just Work"™. Therefore, calling `opend` in your app's folder will look for a folder named "src" or "source" and build every .d file from there.
> Optionally, you can specify the source file directory right after: `opend mysourcedir`. `opend` will guess the output file name by how the current working folder is called.

Reviewer: ```suggestion
The main idea is that calling `opend` should "Just Work"™. Calling `opend` in your program's directory will look for a subdirectory named "src" or "source" and compile every .d file in there.
Optionally, you can specify the source code directory as an argument on the command line, e.g. `opend mysourcedir` to build .d files in the directory "mysourcedir". `opend` will name the output file after the current directory.
```

By default, calling `opend` will execute something like this:
```
dmd src/main.d -od=build/ -of=build/cwd.exe
```

## Options

A few options are available:

> * `--compiler=/path/to/compiler` specifies the path to desired compiler
> * `--output=[library|executable]` specifies the build output type. Default is `executable`
> * `--type=[debug|release]` specifies the build type. Default is `debug`

Reviewer: ```suggestion
* `--compiler=/path/to/compiler` specifies the path to your desired compiler
* `--output=[executable|library]` specifies whether to output an executable or a library (defaults to `executable`)
* `--type=[debug|release]` specifies the build type (defaults to `debug`). This determines whether to include debug symbols or to optimize the code.
```

## Building opend

> Just use `opend` in this directory, duh ～  
> (Also, for the first time, running `dmd src/opend.d -of=opend.exe` should be enough)

Reviewer: ```suggestion
Just call `opend` in this directory. If you don't have it, running `dmd src/opend.d -of=opend` should be enough.
```

## Some future plans

We hope to keep `opend` as simple as possible. Anyway, here's a short list of TODOs:

> * Store build configuration in `.opend` file

Reviewer: ```suggestion
* Store the build configuration in a `.opend` file
```
